src/test/java/com/buyanywhere/productcatalog/controllers/v1/CategoryControllerTests.java

```diff
@@ -7,67 +7,29 @@
 import com.buyanywhere.productcatalog.models.Category;
 import com.buyanywhere.productcatalog.services.CategoriesService;
 import org.junit.Assert;
-import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.mockito.Mockito;
-import static org.mockito.ArgumentMatchers.notNull;
 import org.mockito.internal.util.reflection.FieldSetter;
 import org.modelmapper.ModelMapper;
 import org.springframework.boot.test.mock.mockito.MockBean;
 import org.springframework.test.context.junit4.SpringRunner;
-import java.util.ArrayList;
-import java.util.List;
+
+import static org.mockito.ArgumentMatchers.notNull;
 
 @RunWith(SpringRunner.class)
 public class CategoryControllerTests {
 
     @MockBean
     private CategoriesService service;
 
-    private List<Category> categories;
-
-    @Before
-    public void setup(){
-        categories = new ArrayList<>();
-        Category homeCategory = new Category("Home", 1);
-        categories.add(homeCategory);
-
-        Category newCategory = new Category("NewCategory", 5);
-        categories.add(newCategory);
-
-        Category updateCategory = new Category("UpdateCategory", 7);
-        categories.add(updateCategory);
-
-        Mockito.when(service.exists((long) 1)).thenReturn(true);
-        Mockito.when(service.exists((long) 2)).thenReturn(false);
-        Mockito.when(service.exists((long) 3)).thenReturn(true);
-        Mockito.when(service.exists((long) 4)).thenReturn(true);
-
-        Mockito.when(service.findById((long) 1)).thenReturn(categories.get(0));
-        Mockito.when(service.findById((long) 3)).thenReturn(categories.get(2));
-
-        Mockito.when(service.isDuplicated("DuplicatedName")).thenReturn(true);
-        Mockito.when(service.isDuplicated(newCategory.getName())).thenReturn(false);
-
-        Mockito.when(service.isDuplicated((long) 1, "DuplicatedName")).thenReturn(true);
-
-        try {
-            FieldSetter.setField(homeCategory, homeCategory.getClass().getDeclaredField("id"), (long) 1);
-            FieldSetter.setField(newCategory, newCategory.getClass().getDeclaredField("id"), (long) 2);
-            FieldSetter.setField(updateCategory, updateCategory.getClass().getDeclaredField("id"), (long) 3);
-        } catch (NoSuchFieldException e) {
-            e.printStackTrace();
-        }
-
-        Mockito.when(service.add(notNull())).thenReturn(categories.get(1));
-        Mockito.when(service.update(notNull())).thenReturn(categories.get(2));
-    }
-
     @Test
     public void get_whenExists_shouldReturnCategory(){
-        CategoryController controller = new CategoryController(service, new ModelMapper());
+        Category category = new Category("Home", 1);
+        MockExists(1, true);
+        MockFindById(1, category);
```

Comment:
declare id as finalk long and reuse it. Let's try not to have "magic numbers" spread in the code